sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ExtRefService.java

```diff
@@ -5,41 +5,95 @@
-            .flatMap(List::stream)
-            .collect(Collectors.toList());
-        return SclReport.builder()
-            .sclRootAdapter(sclRootAdapter)
-            .errorDescriptionList(errorDescriptions).build();
+            .map(inputsAdapter -> inputsAdapter.updateAllExtRefIedNames(icdSystemVersionToIed))
+            .flatMap(List::stream).collect(Collectors.toList());
+
+        return new SclReport(sclRootAdapter, extRefErrors);
+    }
+
+    private static List<SclReportItem> validateIed(SclRootAdapter sclRootAdapter) {
+        List<SclReportItem> iedErrors = new ArrayList<>(checkIedUnityOfIcdSystemVersionUuid(sclRootAdapter));
+        iedErrors.addAll(checkIedCompasIcdHeaderAttributes(sclRootAdapter));
+        return iedErrors;
     }
 
+    private static List<SclReportItem> checkIedCompasIcdHeaderAttributes(SclRootAdapter sclRootAdapter) {
+        return sclRootAdapter.streamIEDAdapters()
+            .map(iedAdapter -> {
+                    Optional<TCompasICDHeader> compasPrivate = PrivateService.getCompasPrivate(iedAdapter.getCurrentElem(), TCompasICDHeader.class);
+                    if (compasPrivate.isEmpty()) {
+                        return iedAdapter.buildFatalReportItem(String.format("IED has no Private %s element", PrivateEnum.COMPAS_ICDHEADER.getPrivateType()));
+                    }
+                    if (StringUtils.isBlank(compasPrivate.get().getICDSystemVersionUUID())
+                        || StringUtils.isBlank(compasPrivate.get().getIEDName())) {
+                        return iedAdapter.buildFatalReportItem(String.format("IED private %s as no icdSystemVersionUUID or iedName attribute",
+                            PrivateEnum.COMPAS_ICDHEADER.getPrivateType()));
+                    }
+                    return null;
+                }
+            ).filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    }
 
+    private static List<SclReportItem> checkIedUnityOfIcdSystemVersionUuid(SclRootAdapter sclRootAdapter) {
+        Map<String, List<TIED>> systemVersionToIedList = sclRootAdapter.getCurrentElem().getIED().stream()
+            .collect(Collectors.groupingBy(ied -> PrivateService.getCompasPrivate(ied, TCompasICDHeader.class)
+                .map(TCompasICDHeader::getICDSystemVersionUUID)
+                .orElse("")));
 
+        return systemVersionToIedList.entrySet().stream()
+            .filter(entry -> StringUtils.isNotBlank(entry.getKey()))
+            .filter(entry -> entry.getValue().size() > 1)
+            .map(entry -> SclReportItem.fatal(entry.getValue().stream()
+                .map(tied -> new IEDAdapter(sclRootAdapter, tied))
+                .map(IEDAdapter::getXPath)
+                .collect(Collectors.joining(", ")),
+                "/IED/Private/compas:ICDHeader[@ICDSystemVersionUUID] must be unique" +
+                    " but the same ICDSystemVersionUUID was found on several IED."))
+            .collect(Collectors.toList());
+    }
```

Comment:
this two private method are shared with Scl service importIedAndStd

Reply:
It is not obvious for me how to share these methods with `importIedAndStd`. I suggest creating an issue about that.